x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/EnrichShardMultiSearchAction.java

```diff
@@ -237,7 +237,8 @@ protected MultiSearchResponse shardOperation(Request request, ShardId shardId) t
                     () -> { throw new UnsupportedOperationException(); },
                     null
                 );
-                final Text typeText = context.getMapperService().documentMapper().typeText();
+                final String type = context.getType();
+                final Text typeText = new Text(type);
```

Comment:
I took a shortcut here, to avoid having to expose typeText from MapperService. Is it reasonable?

Reply:
++